src/common/payment.ts

```diff
@@ -182,22 +184,44 @@ export class TurboUnauthenticatedPaymentService
     return addresses;
   }
 
+  protected appendCallbackUrlsToQuery(
+    callbackUrls:
+      | TurboCheckoutSessionHostedParams
+      | TurboCheckoutSessionEmbeddedParams,
+  ): string {
+    return Object.entries(callbackUrls)
+      .filter(([_, value]) => value !== undefined)
+      .map(([key, value]) => `${key}=${encodeURIComponent(value)}`)
+      .join('&');
+  }
+
   protected async getCheckout(
     {
       amount,
       owner,
       promoCodes = [],
       uiMode = 'hosted',
+      ...callbackUrls
     }: TurboCheckoutSessionParams,
     headers?: TurboSignedRequestHeaders,
   ): Promise<TurboCheckoutSessionResponse> {
     const { amount: paymentAmount, type: currencyType } = amount;
 
-    const endpoint = `/top-up/checkout-session/${owner}/${currencyType}/${paymentAmount}?uiMode=${uiMode}${
-      promoCodes.length > 0
-        ? `&${this.appendPromoCodesToQuery(promoCodes)}`
-        : ''
-    }&token=${this.token}`;
+    const queryParams = new URLSearchParams();
+    queryParams.append('uiMode', uiMode);
+    queryParams.append('token', this.token);
+    queryParams.append('promoCode', promoCodes.join(','));
```

Comment:
promoCode is not currently a CSV -- `promoCode=code1&promoCode=code2`

<img width="522" alt="image" src="https://github.com/user-attachments/assets/7ec10538-f787-495b-bf68-c4ad24a68573" />